src/util/artifacts.js

```diff
@@ -1,25 +1,51 @@
 /* @flow */
 import {fs} from 'mz';
+import mkdirp from 'mkdirp';
+import promisify from 'es6-promisify';
 
 import {UsageError, isErrorWithCode} from '../errors';
 import {createLogger} from './logger';
 
 const log = createLogger(__filename);
-
+const asyncMkdirp = promisify(mkdirp);
 
 export async function prepareArtifactsDir(
-  artifactsDir: string
+  artifactsDir: string,
 ): Promise<string> {
   try {
     const stats = await fs.stat(artifactsDir);
     if (!stats.isDirectory()) {
       throw new UsageError(
-        `--artifacts-dir=${artifactsDir} (this value is not a directory)`);
+        `"${artifactsDir}" exists and it is not a directory.`);
+    }
+    // Check that we can write in the artifacts dir.
+    try {
```

Comment:
I would really like to not have so many `try/catch` nested here, but I tried it rewritten without the nested `try/catch` and it didn't look better (to be fair the number of nested `try/catch` is not higher than it was before this change).